internal/pkg/manifest/svc.go

```diff
@@ -73,18 +73,68 @@ func (a *Count) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		}
 	}
 
-	if !a.Autoscaling.IsEmpty() {
+	if !c.Autoscaling.IsValid() {
+		return errUnmarshalSpot
+	}
+
+	if !c.Autoscaling.IsEmpty() {
+		// Successfully unmarshalled Autoscaling fields, return
 		return nil
 	}
 
-	if err := unmarshal(&a.Value); err != nil {
+	if err := unmarshal(&c.Value); err != nil {
 		return errUnmarshalCountOpts
 	}
 	return nil
 }
 
+// Spot is a custom type that can either be represented as a boolean, when
+// specified with a Range, or as an integer, which indicates the `desiredCount`
+// on a service that does not use application autoscaling.
+type Spot struct {
+	Enabled *bool
+	Base    *int
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the Spot
+// struct, allowing it to be unmarshalled into either an int value provided by
+// Base or a boolean value provided by Enabled.
+// This method implements the yaml.Unmarshaler (v2) interface.
+func (s *Spot) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	if err := unmarshal(&s.Base); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if *s.Base != 0 {
+		s.Enabled = nil
+		return nil
+	}
+
+	if err := unmarshal(&s.Enabled); err != nil {
```

Comment:
(If we need to use `if *s.Base != 0` as opposed to `if s.Base != nil` above, then please ignore this comment)

If we no longer need the block that follows (which sets `s.Base = nil`), then we could probably just `return unmarshal(&s.Enabled)`?